analysis/src/python/data_analysis/preprocessing/build_submissions_series.py

```diff
@@ -0,0 +1,113 @@
+import argparse
+import logging
+import sys
+
+import pandas as pd
+
+from analysis.src.python.data_analysis.model.column_name import SubmissionColumns
+from analysis.src.python.data_analysis.utils.df_utils import append_df, write_df
+from analysis.src.python.data_analysis.utils.parsing_utils import str_to_datetime
+
+
+def check_same_code(submission_0: pd.Series, submission_1: pd.Series) -> bool:
+    """ Check if submission not the same as previous. """
+
+    return submission_0[SubmissionColumns.CODE] == submission_1[SubmissionColumns.CODE]
```

Comment:
I think you should use the [strip](https://www.w3schools.com/python/ref_string_strip.asp) function to avoid submissions with new lines and spaces in the start and end of the solution and without any other changes